addons/l10n_jo/models/account_chart_template.py

```diff
@@ -0,0 +1,44 @@
+from odoo import models, _, api
+
+
+class AccountChartTemplate(models.Model):
+    _inherit = 'account.chart.template'
+
+    def _prepare_all_journals(self, acc_template_ref, company, journals_dict=None):
+        """ If JORDAN chart, we add one new journal Tax Adjustment"""
+        if self == self.env.ref('l10n_jo.jordan_chart_template_standard'):
+            if not journals_dict:
+                journals_dict = []
+            journals_dict.extend(
+                [{"name": "Tax Adjustments", "company_id": company.id, "code": "TA", "type": "general", "sequence": 1,
+                  "favorite": True}])
+        return super()._prepare_all_journals(acc_template_ref, company, journals_dict=journals_dict)
+    
+    @api.model
+    def _get_default_bank_journals_data(self):
+        if self == self.env.ref('l10n_jo.jordan_chart_template_standard'):
+            return [
+                {'acc_name': _('Cash'), 'account_type': 'cash', 'default_account_id': self.env.ref('l10n_jo.jo_account_100108')},
+                {'acc_name': _('Bank'), 'account_type': 'bank', 'default_account_id': self.env.ref('l10n_jo.jo_account_100105')}
+            ]
+        return super()._get_default_bank_journals_data()
+    
+    def _create_bank_journals(self, company, acc_template_ref):
+        '''
+        If JORDAN chart, pass default account for bank and cash journal
+        to prevent auto account creation
+        '''
+        if self == self.env.ref('l10n_jo.jordan_chart_template_standard'):
+            self.ensure_one()
+            bank_journals = self.env['account.journal']
+            for acc in self._get_default_bank_journals_data():
+                bank_journals += self.env['account.journal'].create({
+                    'name': acc['acc_name'],
+                    'type': acc['account_type'],
+                    'company_id': company.id,
+                    'currency_id': acc.get('currency_id', self.env['res.currency']).id,
+                    'sequence': 10,
+                    'default_account_id': acc_template_ref.get(acc['default_account_id']).id  # pass default account
+                })
+            return bank_journals
+        return super()._create_bank_journals(company, acc_template_ref)
```

Comment:
```suggestion
```

Reply:
I can't just set the prefix to the required code due to how `_search_new_account_code` is implemented.
It takes the prefix and adds a digit to it and then checks if the account already exists.

So then I tried setting the prefix to `10010`, but then we would need the bank journal to be created first (so that it uses up the 100105) and then the cash journal, taking up the next unused, i.e., 100108.

This is not the case as we see in `_get_default_bank_journals_data`, the Cash journal is created first. Overriding this and switching the order works, so would you rather I do that, like so?

```
def _get_default_bank_journals_data(self):
        if self == self.env.ref('l10n_jo.jordan_chart_template_standard'):
            return super()._get_default_bank_journals_data()[::-1]
        return super()._get_default_bank_journals_data()
```

Reply:
@asfa-odoo And what happens if they want more than 1 bank or cash account?  We want to only have 6 digits all of the time for the account numbers?  @nadaseder @Olivier-olma

Reply:
Hello @jco-odoo 
Yes they should follow the coding approach of the localisation, they can add accounts within the same account group.